porcupine/plugins/directory_tree.py

```diff
@@ -385,12 +385,25 @@ def _on_right_click(self, event: tkinter.Event[DirectoryTree]) -> str | None:
         self.tk.call("focus", self)
 
         item: str = self.identify_row(event.y)
-        self.set_the_selection_correctly(item)
+
+        # Behaviour for a mouse right-click
+        if event.keysym != "App":
+            # Update selected item
+            self.set_the_selection_correctly(item)
+            # Set menu position to cursor position
+            menu_x=event.x_root
+            menu_y=event.y_root
+
+        # Behaviour for menu/application key
+        else:
+            menu_x=self.winfo_rootx()
+            # Vertical menu offset by the focus item's relative position
+            menu_y=self.winfo_rooty()+self.bbox(self.focus())[1]+self.bbox(self.focus())[3]
```

Comment:
- It might be good to offset `menu_x` as well. I think it would look a little better, and it would still work if I later decide to move the directory tree somewhere else than to the left edge of the window.
- While `self.bbox(self.focus())[3]` works, it isn't very readable. Consider something like `x, y, width, height = self.bbox(self.focus())`. This unpacks the returned tuple into separate variables.

Reply:
Thanks for the feedback. Something like, -? I'm not sure how it appears on non-Windows 10 platforms.
```suggestion
            # Get relative position of focus item
            x, y, width, height = self.bbox(self.focus())

            # Apply menu position offsets
            menu_x = int(self.winfo_rootx() + (width / 8))
            menu_y = int(self.winfo_rooty() + y + height)
```

Reply:
We probably don't need to use the width in menu_x at all, because we want the menu to still appear in the left side of the selected item:

```
menu_x = self.winfo_rootx() + x
menu_y = self.winfo_rooty() + y + height
```